### PR TITLE
chore(props): removal of deprecated connect and context APIs

### PR DESCRIPTION
**Pull request by JessicaSachs**
## What is the current behavior?
The Connect and Context APIs both exist and are marked as deprecated.

## What is the new behavior?
We've finally cleaned up and removed the Connect and Context APIs.

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

**Comment by JessicaSachs** [comment]
There's one unit test breaking for the dev build number not matching a particular regexp. Aside from that, the build looks green!

**Review by rwaskiewicz — CHANGES_REQUESTED** [review]
Looks good! 

I had one minor nitpick in the code.

Only blocking ask is to update the git commit message to follow our [semantic commit message format](https://github.com/ionic-team/stencil/blob/main/CONTRIBUTING.md#commit-message-format)

**Review by alicewriteswrongs — APPROVED** [review]
all looks good! always good to delete some code 🗑️

approval pending the section number change @rwaskiewicz mentioned, also looks like this might need a rebase

**Review by rwaskiewicz — APPROVED** [review]

File: BREAKING_CHANGES.md [hunk 1]
```diff
@@ -46,33 +55,91 @@ this field and its corresponding behavior has been removed.
...
-    * [Move `autoDefineCustomElements` Configuration](#move-autodefinecustomelements-configuration)
-    * [Remove `inlineDynamicImports` Configuration](#remove-inlinedynamicimports-configuration)
-  * [`dist-custom-elements-bundle` Output Target](#dist-custom-elements-bundle-output-target)
-* [Legacy Angular Output Target](#legacy-angular-output-target)
-* [Stencil APIs](#stencil-apis)
-  * [Flag Parsing, `parseFlags()`](#flag-parsing-parseflags)
-  * [Destroy Callback, `addDestroy()`, `removeDestroy()`](#destroy-callback-adddestroy-removedestroy)
-* [End-to-End Testing](#end-to-end-testing)
-  * [Puppeteer v10+ Required](#puppeteer-v10-required)
+- [Breaking Changes](#breaking-changes)
```
Comment (tanner-reits) [inline BREAKING_CHANGES.md:58]: Is this a rebase problem or am I missing something? Wouldn't think anything in the v3 section should change

**Review by tanner-reits — CHANGES_REQUESTED** [review]
Changes look good, just one question in the breaking changes md

**Comment by rwaskiewicz** [comment]
@JessicaSachs it looks like the karma tests are now failing - I pulled the branch locally and was able to confirm that this wasn't related to any Browserstack flakiness (IE I was able to repro locally by building this branch locally and with `npm run test.karma.prod` from the root level of the project). Can you PTAL?

**Review by tanner-reits — APPROVED** [review]
Looks good!

**Pull request merged**